index.js

```diff
@@ -12,323 +12,334 @@ const Type = (exports.Type = {
+        if (v.z < minZ) minZ = v.z;
+        if (v.x > maxX) maxX = v.x;
+        if (v.y > maxY) maxY = v.y;
+        if (v.z > maxZ) maxZ = v.z;
+      }
+    });
+    target.min.set(minX, minY, minZ);
+    target.max.set(maxX, maxY, maxZ);
+    return target;
   };
 })();
 
-function _getMeshes(sceneRoot) {
-  let meshes = [];
-  sceneRoot.traverse(o => {
-    if (o.isMesh && (!THREE.Sky || o.__proto__ != THREE.Sky.prototype)) {
-      meshes.push(o);
+const _createBoxShape = function({ x, y, z } = halfExtents) {
+  const btHalfExtents = new Ammo.btVector3(x, y, z);
+  const collisionShape = new Ammo.btBoxShape(btHalfExtents);
+  Ammo.destroy(btHalfExtents);
+  return collisionShape;
+};
+
+const _createCylinderShape = function({ x, y, z } = halfExtents, cylinderAxis) {
+  const btHalfExtents = new Ammo.btVector3(x, y, z);
+  const collisionShape = (() => {
+    switch (cylinderAxis) {
+      case "y":
+        return new Ammo.btCylinderShape(btHalfExtents);
+      case "x":
+        return new Ammo.btCylinderShapeX(btHalfExtents);
+      case "z":
+        return new Ammo.btCylinderShapeZ(btHalfExtents);
     }
-  });
-  return meshes;
-}
-
-const _getVertices = (function() {
-  const vertexPool = [];
-  const vertices = [];
-  const matrix = new THREE.Matrix4();
-  const inverse = new THREE.Matrix4();
-
-  return function(sceneRoot, meshes) {
-    while (vertices.length > 0) {
-      vertexPool.push(vertices.pop());
+    return null;
+  })();
+  Ammo.destroy(btHalfExtents);
+  return collisionShape;
+};
+
+const _createConeShape = function({ x, y, z } = halfExtents, cylinderAxis) {
+  switch (cylinderAxis) {
+    case "y":
+      return new Ammo.btConeShape(Math.max(x, z), y * 2);
+    case "x":
+      return new Ammo.btConeShapeX(Math.max(y, z), x * 2);
+    case "z":
+      return new Ammo.btConeShapeZ(Math.max(x, y), z * 2);
+  }
+  return null;
+};
+
+const _createCapsuleShape = function({ x, y, z } = halfExtents, capsuleAxis) {
+  switch (capsuleAxis) {
+    case "y":
+      return new Ammo.btCapsuleShape(Math.max(x, z), y * 2);
+    case "x":
+      return new Ammo.btCapsuleShapeX(Math.max(y, z), x * 2);
+    case "z":
+      return new Ammo.btCapsuleShapeZ(Math.max(x, y), z * 2);
+  }
+  return null;
+};
+
+const _createHullShape = (function() {
+  const pos = new THREE.Vector3();
+  const quat = new THREE.Quaternion();
+  const scale = new THREE.Vector3();
+  const vertex = new THREE.Vector3();
+  return function(root, mergeGeometry, margin, maxVertices) {
+    const btVertex = new Ammo.btVector3();
+    const originalHull = new Ammo.btConvexHullShape();
+    originalHull.setMargin(margin);
+
+    let vertexCount = 0;
+    _iterateGeometries(root, mergeGeometry, geo => {
+      vertexCount += geo.attributes.position.array.length / 3;
+    });
+
+    // todo: might want to implement this in a deterministic way that doesn't do O(verts) calls to Math.random
+    console.log(`Creating hull shape with ${vertexCount} vertices.`);
```

Comment:
while I think this is potentially useful, I think it needs to be only done in the case that a debug flag is passed in. Going to remove it for now.